### PR TITLE
virttest.utils_conn: Add CA server setup locally

### PR DESCRIPTION
**Pull request by dzhengfy**
Currently TLS connection setup CA server on remote host and CA client on
local host which is to enable the connenction from local host to remote
host. Now there is a need to connect from remote host to local host
using TLS connection. And the expected result is a failure due to no CA
client setup on remote host. So the fix is to only setup CA server on
local host for now.

Signed-off-by: Dan Zheng dzheng@redhat.com

### CODE REVIEW

**Comment by dzhengfy** [comment]
@waynesun09 @kylazhang , could you help a review?

File: virttest/utils_conn.py [hunk 1]
```diff
@@ -1229,6 +1395,63 @@ def build_server_key(tmp_dir, ca_cakey_path=None,
         raise ConnCertError(serverinfo_path, CmdResult.stderr)
 
 
+def build_server_key_on_local(tmp_dir, ca_cakey_path=None,
```
Comment (kylazhang) [inline virttest/utils_conn.py]: Can you reuse the existing server_setup?

**Comment by clebergnu** [comment]
@waynesun09 @kylazhang @dzhengfy any updates here?  Is this PR still relevant?

**Comment by dzhengfy** [comment]
@clebergnu , yes. I will update it recently.

File: virttest/utils_conn.py [hunk 2]
```diff
@@ -1096,6 +1130,138 @@ def server_setup(self):
...
+        # sometimes, need to reuse previous CA cert
+        if self.ca_cakey_path and scp_new_cacert == 'no':
+            cacert_path = '%s/cacert.pem' % self.ca_cakey_path
+            cakey_path = '%s/cakey.pem' % self.ca_cakey_path
+        else:
+            cacert_path = '%s/cacert.pem' % tmp_dir
+            cakey_path = '%s/cakey.pem' % tmp_dir
+
+        serverkey_path = '%s/serverkey.pem' % tmp_dir
+        servercert_path = '%s/servercert.pem' % tmp_dir
```
Comment (balamuruhans) [inline virttest/utils_conn.py]: it is good to use os.path.join() while appending path here and elsewhere

File: virttest/utils_conn.py [hunk 3]
```diff
@@ -1229,6 +1395,63 @@ def build_server_key(tmp_dir, ca_cakey_path=None,
...
+    if not ca_cakey_path:
+        cakey_path = '%s/cakey.pem' % tmp_dir
+        cacert_path = '%s/cacert.pem' % tmp_dir
+    else:
+        cakey_path = '%s/cakey.pem' % ca_cakey_path
+        cacert_path = '%s/cacert.pem' % ca_cakey_path
+
+    serverkey_path = '%s/serverkey.pem' % tmp_dir
+    servercert_path = '%s/servercert.pem' % tmp_dir
+    serverinfo_path = '%s/server.info' % tmp_dir
```
Comment (balamuruhans) [inline virttest/utils_conn.py]: same here

Comment (dzhengfy) [inline virttest/utils_conn.py]: @balamuruhans thanks, updated.

File: virttest/utils_conn.py (hunk 1 above)
Comment (dzhengfy) [inline virttest/utils_conn.py]: @kylazhang I'v updated to reuse. Thanks.

File: virttest/utils_conn.py [hunk 4]
```diff
@@ -964,11 +992,13 @@ def conn_setup(self, server_setup=True, client_setup=True):
             self.server_setup()
         if client_setup:
             self.client_setup()
+        if server_setup_local:
+            self.server_setup(on_local=True)
```
Comment (will-Do) [inline virttest/utils_conn.py]: after this step, there are two servers will be setup, is this expected?

File: virttest/utils_conn.py [hunk 5]
```diff
@@ -964,11 +992,13 @@ def conn_setup(self, server_setup=True, client_setup=True):
...
 
         self.close_session()
         logging.debug("TLS connection setup successfully.")
 
-    def server_setup(self):
+    def server_setup(self, on_local=False):
```
Comment (will-Do) [inline virttest/utils_conn.py]: update docstring

File: virttest/utils_conn.py (hunk 4 above)
Comment (dzhengfy) [reply virttest/utils_conn.py]: yes. This is as expected.

File: virttest/utils_conn.py [hunk 6]
```diff
@@ -1093,30 +1169,50 @@ def server_setup(self):
...
 
         # edit the /etc/libvirt/libvirtd.conf to disable server sanity checks
         if tls_sanity_cert == "no":
             pattern2repl = {r".*tls_no_sanity_certificate\s*=\s*.*":
                             'tls_no_sanity_certificate=1'}
-            self.server_libvirtdconf.sub_else_add(pattern2repl)
+            if on_local:
+                self.client_libvirtdconf.sub_else_add(pattern2repl)
+            else:
+                self.server_libvirtdconf.sub_else_add(pattern2repl)
```
Comment (kylazhang) [inline virttest/utils_conn.py]: So many duplication for this operation, can it be deal with in separate func?

**Comment by dzhengfy** [comment]
@kylazhang Some updates, and did not use separate functions. But I think they are in same complexity.

**Comment by dzhengfy** [comment]
(1/1) type_specific.io-github-autotest-libvirt.virsh.migrate_vm.positive_testing.p2p_migration.basic.with_tls.without_postcopy: PASS (79.38 s)

 (1/1) type_specific.io-github-autotest-libvirt.virsh.migrate_vm.positive_testing.p2p_migration.basic.with_tls_reverse.without_postcopy: PASS (83.18 s)

**Pull request merged**